examples/textures/02-mouse-painting.lisp

```diff
@@ -0,0 +1,176 @@
+                      when swath-under-mouse
+                        return swath-under-mouse))
+
+          (when (and swath-hovered (is-mouse-button-pressed-p +mouse-button-left+))
+            (setf swath-selected swath-hovered
+                  swath-selected-prev swath-selected))
+
+          ;; Clear render texture to clear color
+          (when (is-key-pressed-p +key-c+)
+            (with-texture-mode (target :clear +green+)))
+
+          ;; Update brush size, color & position
+          (let ((radius (min 50 (max 2 (+ (radius brush)
+                                          (* (get-mouse-wheel-move) 5)))))
+                (x (x mouse-pos))
+                (y (y mouse-pos)))
+            (setf (color brush) (color swath-selected))
+            (mapc (lambda (obj) (setf (radius obj) radius
+                                      (x obj) x
+                                      (y obj) y))
+                  (list brush eraser eraser-outline)))
+
+          ;; Paint circle into render texture
+          (when (or (claylib/ll:is-mouse-button-down-p +mouse-button-left+)
+                    (= (claylib/ll:get-gesture-detected) +gesture-drag+))
+            (with-texture-mode (target :clear nil)
+              (when (> (y mouse-pos) 50)
+                (draw-object brush))))
+
+          (if (claylib/ll:is-mouse-button-down-p +mouse-button-right+)
+              (progn
+                (unless mouse-was-pressed
+                  (setf swath-selected-prev swath-selected
+                        swath-selected (first color-swaths)))
+                (setf mouse-was-pressed t)
+
+                ;; Erase circle from render texture
+                (with-texture-mode (target :clear nil)
+                  (when (> (y mouse-pos) 50)
+                    (draw-object eraser))))
+
+              (when (and (claylib/ll:is-mouse-button-released-p +mouse-button-right+)
+                         mouse-was-pressed)
+                (setf swath-selected swath-selected-prev
+                      mouse-was-pressed nil)))
+
+          (with-scene-objects (btn-save-rec btn-save-msg save-rec1 save-rec2 save-text
+                                            white-swath-outline) scene
+            ;; Check if the mouse is hovering over the save button, change color accordingly
+            (let ((color (if (check-collision-point-rec mouse-pos btn-save-rec)
+                             +red+
+                             +black+)))
+              (setf (color btn-save-rec) color
+                    (color btn-save-msg) color))
+
+            ;; TODO Image saving logic
+            (when (or (and btn-save-mouse-hover
+                           (claylib/ll:is-mouse-button-released-p +mouse-button-left+))
+                      (is-key-pressed-p +key-s+))
+              (claylib/ll:load-image-from-texture (c-struct target)))
+
+            (with-drawing
+              ;; TODO Draw render texture
+              ;; (draw-object target)
```

Comment:
The Raylib example line you're trying to replicate is: `DrawTextureRec(target.texture, (Rectangle) { 0, 0, (float)target.texture.width, (float)-target.texture.height }, (Vector2) { 0, 0 }, WHITE);`

So yeah it's not trying to draw `target`, but `target.texture`.